apps/prairielearn/python/prairielearn/core.py

```diff
@@ -290,7 +367,7 @@ def to_json(v, *, df_encoding_version=1, np_encoding_version=1):
         }
 
     if np.isscalar(v) and np.iscomplexobj(v):
-        return {"_type": "complex", "_value": {"real": v.real, "imag": v.imag}}
+        return {"_type": "complex", "_value": {"real": v.real, "imag": v.imag}}  # type: ignore
```

Comment:
Seems like this is because `iscomplexobj` doesn't return `TypeGuard[...]`?

Reply:
![CleanShot 2025-01-26 at 00 27 38@2x](https://github.com/user-attachments/assets/dc05e81b-ac34-4097-a21c-2e30a6bd3906)

Reply:
Yeah, it's a little too generic.